File: Arrays/InsertionSort.cpp
#include <bits/stdc++.h>
using namespace std;

void insertionSort(int arr[], int n)
{
    for(int i=0; i<n; i++)
    {
        int temp = arr[i];
        int j=i;

        while(j>0 && temp<arr[j-1])
        {
            arr[j] = arr[j-1];
            j--;
        }

        arr[j] = temp;
    }
}

File: Arrays/InsertionSort.java
public class InsertionSort {
    public void insertionSort(int[] arr) {
        for (int i = 1; i < arr.length; i++) {
            int cur = arr[i];
            int k = i;

            while (k > 0 && arr[k - 1] > cur) {
                arr[k] = arr[k - 1];
                k--;
            }

            arr[k] = cur;
        }
    }
}
